server/hazard-service/src/main/java/edu/illinois/ncsa/incore/service/hazard/controllers/TornadoController.java

```diff
@@ -264,8 +263,15 @@ public Tornado createTornado(
                 // Store the dataset
                 datasetId = ServiceUtil.createDataset(datasetObject, this.username, this.userGroups, files);
                 // Assuming only one hazardDataset will be created here.
-                TornadoHazardDataset hazardDataset = tornadoModel.getHazardDatasets().get(0);
+                // construct tornado hazardDatasets
+                TornadoHazardDataset hazardDataset = new TornadoHazardDataset();
                 hazardDataset.setDatasetId(datasetId);
+                hazardDataset.setDemandType(TornadoHazard.DEMAND_TYPE);
+                hazardDataset.setDemandUnits(TornadoHazard.WIND_MPH);
+                hazardDataset.setThreshold(null);
+                List<TornadoHazardDataset> hazardDatasetsList = new ArrayList<>();
```

Comment:
I think the TornadoModel would have an empty hazardDatasets variable initialized so maybe you can directly use .addTornadoHazardDataset method?

Reply:
`tornadoModel.addTornadoHazardDataset(hazardDataset)`

Reply:
You are right. addTornadoHazardDataset is probably a better one! I just made the changes

Reply:
Okay! Approving and merging